crates/bevy_pbr/src/bloom/bloom.wgsl

```diff
@@ -0,0 +1,131 @@
+#import bevy_core_pipeline::fullscreen_vertex_shader
+
+struct Uniforms {
+	threshold: f32,
+	knee: f32,
+	scale: f32,
+};
+
+@group(0) @binding(0) var org: texture_2d<f32>;
+@group(0) @binding(1) var org_sampler: sampler;
+@group(0) @binding(2) var<uniform> uniforms: Uniforms;
+@group(0) @binding(3) var up: texture_2d<f32>;
+
+fn quadratic_threshold(color: vec4<f32>, threshold: f32, curve: vec3<f32>) -> vec4<f32> {
+    let br = max(max(color.r, color.g), color.b);
+
+    var rq: f32 = clamp(br - curve.x, 0.0, curve.y);
+    rq = curve.z * rq * rq;
+
+    return color * max(rq, br - threshold) / max(br, 0.0001);
+}
+
+// samples org around the supplied uv using a filter
+//
+// o   o   o
+//   o   o
+// o   o   o
+//   o   o
+// o   o   o
+//
+// this is used because it has a number of advantages that
+// outway the cost of 13 samples that basically boil down
+// to it looking better
+//
+// these advantages are outlined in a youtube video by the Cherno:
+//   https://www.youtube.com/watch?v=tI70-HIc5ro
```

Comment:
Is this the only reference you used? It's a great reference, but just curious if you used anything else.

Reply:
I didn't write this code. I think “Next Generation Post Processing in Call of Duty: Advanced Warfare” is also a good reference iirc.